pkg/apis/kops/validation/validation.go

```diff
@@ -657,24 +658,37 @@ func validateNetworkingCilium(cluster *kops.Cluster, v *kops.CiliumNetworkingSpe
 
 	if v.Version != "" {
 		versionFld := fldPath.Child("version")
-		version, err := semver.ParseTolerant(v.Version)
+		if !strings.HasPrefix(v.Version, "v") {
+			return append(allErrs, field.Invalid(versionFld, v.Version, "Cilium version must be prefixed with 'v'"))
+		}
+		versionString := strings.TrimPrefix(v.Version, "v")
+		version, err := semver.Parse(versionString)
 
 		version.Pre = nil
 		version.Build = nil
 		if err != nil {
 			allErrs = append(allErrs, field.Invalid(versionFld, v.Version, "Could not parse as semantic version"))
 		}
 
-		v8, _ := semver.Parse("1.8.0")
-		v7, _ := semver.Parse("1.7.0")
-		v6, _ := semver.Parse("1.6.0")
+		if !(version.Minor >= 6 && version.Minor <= 8) {
+			allErrs = append(allErrs, field.Invalid(versionFld, v.Version, "Only versions 1.6 through 1.8 are supported"))
+		}
+
+		if version.Minor == 6 && (!cluster.IsKubernetesGTE("1.11") || cluster.IsKubernetesGTE("1.16")) {
+			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.6 requires kubernetesVersion between 1.11 and 1.16"))
+		}
 
-		if !(version.GTE(v6) && version.LT(v8)) {
-			allErrs = append(allErrs, field.Invalid(versionFld, v.Version, "Only versions 1.6 and 1.7 are supported"))
+		if version.Minor == 7 && (!cluster.IsKubernetesGTE("1.12") || cluster.IsKubernetesGTE("1.17")) {
+			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.7 requires kubernetesVersion between 1.12 and 1.17"))
+		}
+		if version.Minor == 8 && (!cluster.IsKubernetesGTE("1.12") || (cluster.IsKubernetesGTE("1.18") && os.Getenv("KOPS_RUN_TOO_NEW_VERSION") == "")) {
```

Comment:
I doubt this won't work once Kubernetes 1.19 is released.
Also think about using or adding `cluster.IsKubernetesLT`.
```suggestion
		if version.Minor == 8 && !cluster.IsKubernetesGTE("1.12") {
```

Reply:
You are probably correct. I also see that the cilium docs have changed the wording slightly: https://docs.cilium.io/en/v1.8/concepts/kubernetes/requirements/#kubernetes-version